File: cpp/Ice/context/ContextI.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include "ContextI.h"
#include <Ice/Ice.h>
#include <iostream>

using namespace std;

void
ContextI::call(const Ice::Current& c)
{
    cout << "Type = ";
    const auto p = c.ctx.find("type");
    cout << (p != c.ctx.end() ? p->second : "None") << endl;
}

void
ContextI::shutdown(const Ice::Current& c)
{
    cout << "Shutting down..." << endl;
    c.adapter->getCommunicator()->shutdown();
}

File: java/Ice/context/ContextI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import com.zeroc.demos.Ice.context.Demo.*;

public class ContextI implements Context
{
    @Override
    public void call(com.zeroc.Ice.Current current)
    {
        System.out.print("Type = ");
        String type = current.ctx.get("type");
        if(type != null)
        {
            System.out.print(type);
        }
        else
        {
            System.out.print("None");
        }
        System.out.println();
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        System.out.println("Shutting down...");
        current.adapter.getCommunicator().shutdown();
    }
}
